packages/xforms-engine/src/lib/reactivity/text/createTextRange.ts

```diff
@@ -74,10 +138,18 @@ export const createTextRange = <Role extends TextRole>(
 	definition: TextRangeDefinition<Role>
 ): ComputedFormTextRange<Role> => {
 	return context.scope.runTask(() => {
-		const getTextChunks = createTextChunks(context, definition.chunks);
+		const textChunks = createTextChunks(context, definition.chunks);
+		const evaluator = isEngineXPathEvaluator(context.evaluator) ? context.evaluator : null;
```

Comment:
The type static-check enforces this validation. I tested several forms (all question types form, all note form, all select form, etc). When creating a Text Range, the evaluator always has a root node that is a Primary Instance. 
But, to play safe, we are leaving this validation.

Reply:
The evaluator, in this case, is only used for retrieving media attachments. I've added it as part of the media context, so its role is clearer in the code.